discover/discover.go

```diff
@@ -80,26 +80,23 @@ func ScanAndConnect(host string, username string, password string, options ...Op
 	for _, probeID := range order {
 		probeDevice := devices[probeID]
 
-		opts.Logger.V(1).Info("probing to identify device", "step", "ScanAndConnect", "host", host)
+		opts.Logger.V(1).Info("probing to identify device", "step", "ScanAndConnect", "host", host, "vendor", probeID)
 
-		bmcConnection, err := probeDevice(opts.Context, opts.Logger)
+		bmcConnection, probeErr := probeDevice(opts.Context, opts.Logger)
```

Comment:
I'm not sure it's necessary to have separate error variables, I believe it's typically common practice to re-use if there's no overlap.

Reply:
sure, will go back to `err`.